{
  "url": "https://api.github.com/repos/ledgerwatch/erigon/issues/6684",
  "repository_url": "https://api.github.com/repos/ledgerwatch/erigon",
  "labels_url": "https://api.github.com/repos/ledgerwatch/erigon/issues/6684/labels{/name}",
  "comments_url": "https://api.github.com/repos/ledgerwatch/erigon/issues/6684/comments",
  "events_url": "https://api.github.com/repos/ledgerwatch/erigon/issues/6684/events",
  "html_url": "https://github.com/ledgerwatch/erigon/issues/6684",
  "id": 1554666502,
  "node_id": "I_kwDOC0FsAM5cqlQG",
  "number": 6684,
  "title": "`torrent.upload.rate` and `torrent.download.rate` seem not to have effect on network bandwidth usage",
  "user": {
    "login": "artur-jablonski",
    "id": 29547623,
    "node_id": "MDQ6VXNlcjI5NTQ3NjIz",
    "avatar_url": "https://avatars.githubusercontent.com/u/29547623?v=4",
    "gravatar_id": "",
    "url": "https://api.github.com/users/artur-jablonski",
    "html_url": "https://github.com/artur-jablonski",
    "followers_url": "https://api.github.com/users/artur-jablonski/followers",
    "following_url": "https://api.github.com/users/artur-jablonski/following{/other_user}",
    "gists_url": "https://api.github.com/users/artur-jablonski/gists{/gist_id}",
    "starred_url": "https://api.github.com/users/artur-jablonski/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/artur-jablonski/subscriptions",
    "organizations_url": "https://api.github.com/users/artur-jablonski/orgs",
    "repos_url": "https://api.github.com/users/artur-jablonski/repos",
    "events_url": "https://api.github.com/users/artur-jablonski/events{/privacy}",
    "received_events_url": "https://api.github.com/users/artur-jablonski/received_events",
    "type": "User",
    "site_admin": false
  },
  "labels": [

  ],
  "state": "closed",
  "locked": false,
  "assignee": null,
  "assignees": [

  ],
  "milestone": null,
  "comments": 3,
  "created_at": "2023-01-24T10:00:37Z",
  "updated_at": "2023-01-29T14:26:25Z",
  "closed_at": "2023-01-29T14:26:24Z",
  "author_association": "NONE",
  "active_lock_reason": null,
  "body": "#### System information\r\n\r\nErigon version: `v2.30.0`\r\n\r\nOS & Version: Linux\r\n\r\nCommit hash : 35f493c5d021965f994bee615d441b2b35811f8f\r\n\r\n#### Expected behaviour\r\nWhen erigon is started with `torrent.upload.rate` and `torrent.download.rate` the overall bandwidth usage doesn't exceed the tresholds specified\r\n\r\n#### Actual behaviour\r\n\r\n`torrent.upload.rate` and `torrent.download.rate` seem not to have effect on bandwidth consumption\r\n\r\n\r\n#### Steps to reproduce the behaviour\r\nstart erigon with `--torrent.upload.rate=1mb` and `--torrent.download.rate=1mb`\r\n\r\n#### Backtrace\r\n\r\nThis is bandwidth shown by `nload` tool. The upload goes to 16MBit/s and download to 3MBit/s\r\n\r\n![bandwidth](https://user-images.githubusercontent.com/29547623/214262544-088828ce-d7e5-491e-a7aa-d4c0c6183a3e.png)\r\n\r\nUnless I am reading nload output wrong, it's a bit unexpected to see such high upload rate from erigon. I tried to throttle on the interface with `tc` to 1Mb/s , but then nothing seems to be synchronizing, the logs just keep showing this forever.\r\n\r\n\r\n```\r\nINFO[01-24|14:07:46.447] [p2p] GoodPeers                          eth66=0\r\nINFO[01-24|14:07:51.193] [Checkpoint Sync] Retrieving bootstrap   peers=0\r\nINFO[01-24|14:08:01.194] [Checkpoint Sync] Retrieving bootstrap   peers=0\r\nINFO[01-24|14:08:11.193] [Checkpoint Sync] Retrieving bootstrap   peers=0\r\nINFO[01-24|14:08:21.194] [Checkpoint Sync] Retrieving bootstrap   peers=0\r\nINFO[01-24|14:08:31.193] [Checkpoint Sync] Retrieving bootstrap   peers=0\r\nINFO[01-24|14:08:41.194] [Checkpoint Sync] Retrieving bootstrap   peers=0\r\nINFO[01-24|14:08:51.193] [Checkpoint Sync] Retrieving bootstrap   peers=0\r\nINFO[01-24|14:09:01.193] [Checkpoint Sync] Retrieving bootstrap   peers=0\r\nINFO[01-24|14:09:11.193] [Checkpoint Sync] Retrieving bootstrap   peers=0\r\nINFO[01-24|14:09:21.194] [Checkpoint Sync] Retrieving bootstrap   peers=0\r\nINFO[01-24|14:09:31.194] [Checkpoint Sync] Retrieving bootstrap   peers=0\r\nINFO[01-24|14:09:41.194] [Checkpoint Sync] Retrieving bootstrap   peers=0\r\nINFO[01-24|14:09:46.446] [p2p] GoodPeers                          eth66=0\r\nINFO[01-24|14:09:51.193] [Checkpoint Sync] Retrieving bootstrap   peers=0\r\nINFO[01-24|14:10:01.194] [Checkpoint Sync] Retrieving bootstrap   peers=0\r\nINFO[01-24|14:10:11.194] [Checkpoint Sync] Retrieving bootstrap   peers=0\r\nINFO[01-24|14:10:21.194] [Checkpoint Sync] Retrieving bootstrap   peers=0\r\nINFO[01-24|14:10:31.194] [Checkpoint Sync] Retrieving bootstrap   peers=0\r\nINFO[01-24|14:10:41.193] [Checkpoint Sync] Retrieving bootstrap   peers=0\r\nINFO[01-24|14:10:51.193] [Checkpoint Sync] Retrieving bootstrap   peers=0\r\nINFO[01-24|14:11:01.193] [Checkpoint Sync] Retrieving bootstrap   peers=0\r\nINFO[01-24|14:11:11.194] [Checkpoint Sync] Retrieving bootstrap   peers=0\r\nINFO[01-24|14:11:21.193] [Checkpoint Sync] Retrieving bootstrap   peers=0\r\nINFO[01-24|14:11:31.194] [Checkpoint Sync] Retrieving bootstrap   peers=0\r\nINFO[01-24|14:11:41.193] [Checkpoint Sync] Retrieving bootstrap   peers=0\r\nINFO[01-24|14:11:46.447] [p2p] GoodPeers                          eth66=0\r\n```\r\n\r\nThis is running on 50Mbit/s down and 20Mbit/s up connection. When it goes full speed it saturates the outgoing link and other devices on the network can't access external world.\r\n\r\n\r\n\r\n",
  "closed_by": {
    "login": "artur-jablonski",
    "id": 29547623,
    "node_id": "MDQ6VXNlcjI5NTQ3NjIz",
    "avatar_url": "https://avatars.githubusercontent.com/u/29547623?v=4",
    "gravatar_id": "",
    "url": "https://api.github.com/users/artur-jablonski",
    "html_url": "https://github.com/artur-jablonski",
    "followers_url": "https://api.github.com/users/artur-jablonski/followers",
    "following_url": "https://api.github.com/users/artur-jablonski/following{/other_user}",
    "gists_url": "https://api.github.com/users/artur-jablonski/gists{/gist_id}",
    "starred_url": "https://api.github.com/users/artur-jablonski/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/artur-jablonski/subscriptions",
    "organizations_url": "https://api.github.com/users/artur-jablonski/orgs",
    "repos_url": "https://api.github.com/users/artur-jablonski/repos",
    "events_url": "https://api.github.com/users/artur-jablonski/events{/privacy}",
    "received_events_url": "https://api.github.com/users/artur-jablonski/received_events",
    "type": "User",
    "site_admin": false
  },
  "reactions": {
    "url": "https://api.github.com/repos/ledgerwatch/erigon/issues/6684/reactions",
    "total_count": 0,
    "+1": 0,
    "-1": 0,
    "laugh": 0,
    "hooray": 0,
    "confused": 0,
    "heart": 0,
    "rocket": 0,
    "eyes": 0
  },
  "timeline_url": "https://api.github.com/repos/ledgerwatch/erigon/issues/6684/timeline",
  "performed_via_github_app": null,
  "state_reason": "completed"
}
[
  {
    "url": "https://api.github.com/repos/ledgerwatch/erigon/issues/comments/1405714591",
    "html_url": "https://github.com/ledgerwatch/erigon/issues/6684#issuecomment-1405714591",
    "issue_url": "https://api.github.com/repos/ledgerwatch/erigon/issues/6684",
    "id": 1405714591,
    "node_id": "IC_kwDOC0FsAM5TyYCf",
    "user": {
      "login": "artur-jablonski",
      "id": 29547623,
      "node_id": "MDQ6VXNlcjI5NTQ3NjIz",
      "avatar_url": "https://avatars.githubusercontent.com/u/29547623?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/artur-jablonski",
      "html_url": "https://github.com/artur-jablonski",
      "followers_url": "https://api.github.com/users/artur-jablonski/followers",
      "following_url": "https://api.github.com/users/artur-jablonski/following{/other_user}",
      "gists_url": "https://api.github.com/users/artur-jablonski/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/artur-jablonski/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/artur-jablonski/subscriptions",
      "organizations_url": "https://api.github.com/users/artur-jablonski/orgs",
      "repos_url": "https://api.github.com/users/artur-jablonski/repos",
      "events_url": "https://api.github.com/users/artur-jablonski/events{/privacy}",
      "received_events_url": "https://api.github.com/users/artur-jablonski/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2023-01-26T21:51:40Z",
    "updated_at": "2023-01-26T21:51:40Z",
    "author_association": "NONE",
    "body": "Hmm, today I looked at this again, started erigon v2.30.0 with --torrent.upload.rate=1mb and --torrent.download.rate=1mb and it seems to be holding the limits:\r\n\r\n![image](https://user-images.githubusercontent.com/29547623/214958592-5f52207f-6b66-4f0c-a8f4-eea2bbefa44e.png)\r\n\r\n\r\nThe logs now are showing this:\r\n\r\n![image](https://user-images.githubusercontent.com/29547623/214958399-fe149bbe-a3d4-421d-b7d3-6a13cbe07499.png)\r\n\r\nNot sure what happened before..... ",
    "reactions": {
      "url": "https://api.github.com/repos/ledgerwatch/erigon/issues/comments/1405714591/reactions",
      "total_count": 0,
      "+1": 0,
      "-1": 0,
      "laugh": 0,
      "hooray": 0,
      "confused": 0,
      "heart": 0,
      "rocket": 0,
      "eyes": 0
    },
    "performed_via_github_app": null
  },
  {
    "url": "https://api.github.com/repos/ledgerwatch/erigon/issues/comments/1405731951",
    "html_url": "https://github.com/ledgerwatch/erigon/issues/6684#issuecomment-1405731951",
    "issue_url": "https://api.github.com/repos/ledgerwatch/erigon/issues/6684",
    "id": 1405731951,
    "node_id": "IC_kwDOC0FsAM5TycRv",
    "user": {
      "login": "artur-jablonski",
      "id": 29547623,
      "node_id": "MDQ6VXNlcjI5NTQ3NjIz",
      "avatar_url": "https://avatars.githubusercontent.com/u/29547623?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/artur-jablonski",
      "html_url": "https://github.com/artur-jablonski",
      "followers_url": "https://api.github.com/users/artur-jablonski/followers",
      "following_url": "https://api.github.com/users/artur-jablonski/following{/other_user}",
      "gists_url": "https://api.github.com/users/artur-jablonski/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/artur-jablonski/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/artur-jablonski/subscriptions",
      "organizations_url": "https://api.github.com/users/artur-jablonski/orgs",
      "repos_url": "https://api.github.com/users/artur-jablonski/repos",
      "events_url": "https://api.github.com/users/artur-jablonski/events{/privacy}",
      "received_events_url": "https://api.github.com/users/artur-jablonski/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2023-01-26T22:05:13Z",
    "updated_at": "2023-01-26T22:05:13Z",
    "author_association": "NONE",
    "body": "Ok, it seems that when erigon is stuck here:\r\n\r\n```\r\nINFO[01-26|21:59:11.422] [Checkpoint Sync] Retrieving bootstrap   peers=0\r\nINFO[01-26|21:59:21.422] [Checkpoint Sync] Retrieving bootstrap   peers=0\r\nINFO[01-26|21:59:31.421] [Checkpoint Sync] Retrieving bootstrap   peers=0\r\n```\r\n\r\nthe network usage goes above thresholds and seems that it cannot unlock itself. Need to bounce it and once it gets over that part, then the limits are working... does it make any sense to anyone?",
    "reactions": {
      "url": "https://api.github.com/repos/ledgerwatch/erigon/issues/comments/1405731951/reactions",
      "total_count": 0,
      "+1": 0,
      "-1": 0,
      "laugh": 0,
      "hooray": 0,
      "confused": 0,
      "heart": 0,
      "rocket": 0,
      "eyes": 0
    },
    "performed_via_github_app": null
  },
  {
    "url": "https://api.github.com/repos/ledgerwatch/erigon/issues/comments/1407677986",
    "html_url": "https://github.com/ledgerwatch/erigon/issues/6684#issuecomment-1407677986",
    "issue_url": "https://api.github.com/repos/ledgerwatch/erigon/issues/6684",
    "id": 1407677986,
    "node_id": "IC_kwDOC0FsAM5T53Yi",
    "user": {
      "login": "artur-jablonski",
      "id": 29547623,
      "node_id": "MDQ6VXNlcjI5NTQ3NjIz",
      "avatar_url": "https://avatars.githubusercontent.com/u/29547623?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/artur-jablonski",
      "html_url": "https://github.com/artur-jablonski",
      "followers_url": "https://api.github.com/users/artur-jablonski/followers",
      "following_url": "https://api.github.com/users/artur-jablonski/following{/other_user}",
      "gists_url": "https://api.github.com/users/artur-jablonski/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/artur-jablonski/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/artur-jablonski/subscriptions",
      "organizations_url": "https://api.github.com/users/artur-jablonski/orgs",
      "repos_url": "https://api.github.com/users/artur-jablonski/repos",
      "events_url": "https://api.github.com/users/artur-jablonski/events{/privacy}",
      "received_events_url": "https://api.github.com/users/artur-jablonski/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2023-01-29T14:26:24Z",
    "updated_at": "2023-01-29T14:26:24Z",
    "author_association": "NONE",
    "body": "Running version v2.37.0. It's the first version that runs full speed and is not  causing knocking-off effect on other devices on the network. Closing this one.",
    "reactions": {
      "url": "https://api.github.com/repos/ledgerwatch/erigon/issues/comments/1407677986/reactions",
      "total_count": 0,
      "+1": 0,
      "-1": 0,
      "laugh": 0,
      "hooray": 0,
      "confused": 0,
      "heart": 0,
      "rocket": 0,
      "eyes": 0
    },
    "performed_via_github_app": null
  }
]
